### PR TITLE
[EET-1918,EET-2143] add preflight-trigger task

### PR DESCRIPTION
**Pull request by komish**
This is an initial task that utilizes preflight-trigger to start a preflight execution in DPTP. This won't work out of the box because it requires specific credentials, etc., so it has not been added to any of the pipelines in this PR. Please also see the mentioned Jira EET-1918 for some decisions we'll need to make before we start using this.

Signed-off-by: Jose R. Gonzalez <josegonzalez89@gmail.com>

cc @mrhillsman

### CODE REVIEW

File: ansible/roles/operator-pipeline/templates/openshift/tasks/preflight-trigger.yaml
Comment (MarcinGinszt) [inline ansible/roles/operator-pipeline/templates/openshift/tasks/preflight-trigger.yaml:54]: Why do we need Kubeconfig? How can we supply it- to which cluster will it point?
We expected this task to output the Kubeconfig so that we could connect to some cluster and run the preflight there.

Comment (komish) [reply ansible/roles/operator-pipeline/templates/openshift/tasks/preflight-trigger.yaml:54]: This is the kubeconfig preflight-trigger uses to apply the prowjob to the DPTP cluster - we'll discuss how to get that over to you.

**Comment by jpaulovic** [comment]
@komish @MarcinGinszt @amisstea I haven't seen any activity on this PR since 8 days ago. Is there anything remaining to do before this can be merged? Also, are you going to add the necessary parameters to the pipeline config or are you expecting us to do that? (both is fine, we're just blocked by this right now)

**Comment by komish** [comment]
/do-not-merge hold

@jpaulovic This is just waiting for the work in #79 to finish. I don't think either one contains the pipeline config, but we'll add that as well.

**Comment by komish** [comment]
@jpaulovic @amisstea This is ready for review, and includes the pipeline changes.

This PR will:

- Add a preflight-trigger task, which runs Preflight in DPTP. This then pulls encrypted results from the DPTP Execution, decrypts them, and places them in the associated "results" locations for the upload task. 
- Rewires the Preflight task in the hosted pipeline to use preflight-trigger as the taskRef instead of preflight.

**Review by haripate — APPROVED** [review]
LGTM

File: ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-hosted-pipeline.yml
Comment (kosciCZ) [inline ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-hosted-pipeline.yml:451]: AFAIK the preflight should run for the latest supported version of the operator:

```
        - name: ocp_version
          value: "$(tasks.get-supported-versions.results.max_supported_ocp_version)"
```

File: ansible/roles/operator-pipeline/templates/openshift/tasks/preflight-trigger.yml
Comment (kosciCZ) [inline ansible/roles/operator-pipeline/templates/openshift/tasks/preflight-trigger.yml:7]: I'd probably avoid setting a default here, I think we'll always supply it (see my other comment)

**Review by kosciCZ — CHANGES_REQUESTED** [review]

File: ansible/roles/operator-pipeline/templates/openshift/tasks/preflight-trigger.yml
Comment (mrhillsman) [reply ansible/roles/operator-pipeline/templates/openshift/tasks/preflight-trigger.yml:7]: Yes we are not going to set the at all and are going to depend on the max_supported_ocp_version as you noted in your previous comment. It is hardcoded for now to ensure we run against a version dptp definitely has in their Hive clusterpool.

File: ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-hosted-pipeline.yml
Comment (mrhillsman) [reply ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-hosted-pipeline.yml:451]: See https://github.com/redhat-openshift-ecosystem/operator-pipelines/pull/67#discussion_r719554484

**Comment by komish** [comment]
@mrhillsman per request - the task name has changed to be preflight-trigger, and result references have been updated to match.

File: ansible/roles/operator-pipeline/templates/openshift/tasks/preflight-trigger.yml
Comment (komish) [reply ansible/roles/operator-pipeline/templates/openshift/tasks/preflight-trigger.yml:7]: Indeed. I think we need to land on a list of supported versions _today_, after which @mrhillsman and I can make sure DPTP has the right configurables that preflight-trigger can reach into before we remove this default value.

**Review by kosciCZ — APPROVED** [review]

**Pull request merged**
